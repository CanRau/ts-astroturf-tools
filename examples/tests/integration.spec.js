const path = require('path');
const runner = require('./runner');

jest.setTimeout(30000);

describe('integration tests', () => {
  describe('typescript-babel', () => {
    it('completes successfully and shows a warning', async () => {
      const workingDirectory = path.resolve(__dirname, '../typescript-babel');
      await expect(runner.run(workingDirectory, 'yarn', [])).resolves.toContain(
        `Identifier "classB" is unused.`
      );
    });
  });

  describe('typescript-webpack', () => {
    it('completes successfully and shows a warning', async () => {
      const workingDirectory = path.resolve(__dirname, '../typescript-webpack');
      await expect(runner.run(workingDirectory, 'yarn', [])).resolves.toContain(
        `Identifier "classB" is unused.`
      );
    });
  });
});
