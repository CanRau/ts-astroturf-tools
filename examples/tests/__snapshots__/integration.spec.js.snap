// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration tests typescript-raw basic snapshot test - styled-remote-interpolation 1`] = `
"import styled from 'astroturf';
import { RED, GREEN } from './utils/colors';
export const SomeComponent = styled.div \`
  color: \${\\"red\\"};
  background: \${\\"green\\"};
\`;
//# sourceMappingURL=styled-remote-interpolation.js.map"
`;

exports[`integration tests typescript-raw basic snapshot test - xcss 1`] = `
"import { xcss } from '../../../xcss';
export const someClassName = (css \`
.someClassName {

  color: red;

}\`).someClassName;
//# sourceMappingURL=xcss.js.map"
`;

exports[`integration tests typescript-raw basic snapshot test - xcss-local-interpolation 1`] = `
"import { xcss } from '../../../xcss';
const RED = 'red';
export const someClassName = (css \`
.someClassName {

  color: \${RED};

}
\`).someClassName;
//# sourceMappingURL=xcss-local-interpolation.js.map"
`;

exports[`integration tests typescript-raw basic snapshot test - xcss-property-assignment 1`] = `
"import { xcss } from '../../../xcss';
export const classes = {
    someClassName: (function () {
        var p_someClassName = css \`
.someClassName {

    color: red;
  
}\`;
        return p_someClassName.someClassName;
    })(),
};
//# sourceMappingURL=xcss-property-assignment.js.map"
`;

exports[`integration tests typescript-raw basic snapshot test - xcss-property-assignment-local-interpolation 1`] = `
"import { xcss } from '../../../xcss';
const RED = 'red';
export const classes = {
    someClassName: (function () {
        var p_someClassName = css \`
.someClassName {

    color: \${RED};
  
}
\`;
        return p_someClassName.someClassName;
    })(),
};
//# sourceMappingURL=xcss-property-assignment-local-interpolation.js.map"
`;

exports[`integration tests typescript-raw basic snapshot test - xcss-property-assignment-remote-interpolation 1`] = `
"import { RED, GREEN } from './utils/colors';
import { xcss } from '../../../xcss';
export const classes = {
    someClassName: (function () {
        var p_someClassName = css \`
.someClassName {

    color: \${\\"red\\"};
    background: \${\\"green\\"};
  
}
\`;
        return p_someClassName.someClassName;
    })(),
};
//# sourceMappingURL=xcss-property-assignment-remote-interpolation.js.map"
`;

exports[`integration tests typescript-raw basic snapshot test - xcss-remote-interpolation 1`] = `
"import { RED, GREEN } from './utils/colors';
import { xcss } from '../../../xcss';
export const someClassName = (css \`
.someClassName {

  color: \${\\"red\\"};
  background: \${\\"green\\"};

}
\`).someClassName;
//# sourceMappingURL=xcss-remote-interpolation.js.map"
`;

exports[`integration tests typescript-webpack basic snapshot test - css 1`] = `
"
  .css-null-astroturf-local_classA {
    color: red;
  }

  .css-null-astroturf-local_classB {
    color: green;
  }


/*# sourceMappingURL=style.css.map*/"
`;

exports[`integration tests typescript-webpack basic snapshot test - css-local-interpolation 1`] = `
"
  .css-local-interpolation-null-astroturf-local_classA {
    color: red;
  }

  .css-local-interpolation-null-astroturf-local_classB {
    color: green;
  }


/*# sourceMappingURL=style.css.map*/"
`;

exports[`integration tests typescript-webpack basic snapshot test - styled 1`] = `
".styled-SomeComponent-astroturf-local_cls1 {
color: red;
}

/*# sourceMappingURL=style.css.map*/"
`;

exports[`integration tests typescript-webpack basic snapshot test - styled-local-interpolation 1`] = `
".styled-local-interpolation-SomeComponent-astroturf-local_cls1 {
color: red;
}

/*# sourceMappingURL=style.css.map*/"
`;

exports[`integration tests typescript-webpack basic snapshot test - styled-remote-interpolation 1`] = `
".styled-remote-interpolation-SomeComponent-astroturf-local_cls1 {
color: red;
background: green;
}

/*# sourceMappingURL=style.css.map*/"
`;

exports[`integration tests typescript-webpack basic snapshot test - xcss 1`] = `
"
.xcss-someClassName-astroturf-local_someClassName {

  color: red;

}

/*# sourceMappingURL=style.css.map*/"
`;

exports[`integration tests typescript-webpack basic snapshot test - xcss-local-interpolation 1`] = `
"
.xcss-local-interpolation-someClassName-astroturf-local_someClassName {

  color: red;

}


/*# sourceMappingURL=style.css.map*/"
`;

exports[`integration tests typescript-webpack basic snapshot test - xcss-property-assignment 1`] = `
"
.xcss-property-assignment-p_someClassName-astroturf-local_someClassName {

    color: red;
  
}

/*# sourceMappingURL=style.css.map*/"
`;

exports[`integration tests typescript-webpack basic snapshot test - xcss-property-assignment-local-interpolation 1`] = `
"
.xcss-property-assignment-local-interpolation-p_someClassName-astroturf-local_someClassName {

    color: red;
  
}


/*# sourceMappingURL=style.css.map*/"
`;

exports[`integration tests typescript-webpack basic snapshot test - xcss-property-assignment-remote-interpolation 1`] = `
"
.xcss-property-assignment-remote-interpolation-p_someClassName-astroturf-local_someClassName {

    color: red;
    background: green;
  
}


/*# sourceMappingURL=style.css.map*/"
`;

exports[`integration tests typescript-webpack basic snapshot test - xcss-remote-interpolation 1`] = `
"
.xcss-remote-interpolation-someClassName-astroturf-local_someClassName {

  color: red;
  background: green;

}


/*# sourceMappingURL=style.css.map*/"
`;
